runtime/hydradx/src/xcm.rs

```diff
@@ -309,3 +309,92 @@ pub type LocalAssetTransactor = ReroutingMultiCurrencyAdapter<
 	OmnipoolProtocolAccount,
 	TreasuryAccount,
 >;
+
+/// A call filter for the XCM Transact instruction. This is a temporary measure until we properly
+/// account for proof size weights.
+///
+/// Calls that are allowed through this filter must:
+/// 1. Have a fixed weight;
+/// 2. Cannot lead to another call being made;
+/// 3. Have a defined proof size weight, e.g. no unbounded vecs in call parameters.
+pub struct SafeCallFilter;
+impl Contains<RuntimeCall> for SafeCallFilter {
+	fn contains(call: &RuntimeCall) -> bool {
+		#[cfg(feature = "runtime-benchmarks")]
+		{
+			if matches!(call, RuntimeCall::System(frame_system::Call::remark_with_event { .. })) {
+				return true;
+			}
+		}
+
+		// check the runtime call filter
+		if !CallFilter::contains(call) {
+			return false;
+		}
+
+		matches!(
```

Comment:
I think this is very extreme. This will disallow some use-cases which we already "support". (Like providing liq)

Reply:
Oh... It's actually allowed list right? Very confusing with the thing above !

Reply:
Yes. The extrinsics in the macro are allowed.